api/envoy/api/v2/route/route.proto

```diff
@@ -39,17 +39,21 @@ message VirtualHost {
   string name = 1 [(validate.rules).string.min_bytes = 1];
 
   // A list of domains (host/authority header) that will be matched to this
-  // virtual host. Wildcard hosts are supported in the form of ``*.foo.com`` or
-  // ``*-bar.foo.com``.
+  // virtual host. Wildcard hosts are supported in the suffix or prefix form.
+  //
+  // Domain search order:
+  //  1. Exact domain names: ``www.foo.com``.
+  //  2. Suffix domain wildcards: ``*.foo.com`` or ``*-bar.foo.com``.
+  //  3. Prefix domain wildcards: ``foo.*`` or ``foo-*``.
```

Comment:
How does this work with SNI? Presumably if we have wildcard certs, they won't work as prefix domains, so you will need to have exact certs or suffix domain certs with SAN. That's a completely different part of Envoy, but I'm wondering operationally whether prefix domain matching will only really work for non-TLS downstream.

Reply:
@htuch AFAIK prefix wildcard certs will/should not work with SNI. Although it is technically possible to create a certificate with a name, say, `mail.*`, I doubt any trusted CA would issue such certificate due to security reasons. I could imagine such scenario with own CA only. I have made such experiment using nginx/curl to no success.

Having said that I think SNI and HTTP domain matching are quite loosely coupled. One can have listener(s) configured with or without suffix wildcards, with or without TLS and at the same time match vhosts with prefix wildcards. It depends on what suits best given scenario.

Reply:
Sure, I agree on the loose coupling, just wondering out loud about how an operator would make use of these two features in conjunction. Nothing actionable, so I will resolve this thread.